lib/i18n/backend/active_record.rb

```diff
@@ -45,6 +45,33 @@ def store_translations(locale, data, options = {})
           end
         end
 
+        def reload!
+          @initialized = false
+          @translations = nil
+        end
+
+        def initialized?
+          @initialized ||= false
+        end
+
+        def init_translations
+          @translations = Translation.all.each.with_object({}) do |t, memo|
+            locale_hash = (memo[t.locale.to_sym] ||= {})
+            keys = t.key.split('.')
+            keys.each.with_index.inject(locale_hash) do |iterator, (key_part, index)|
+              key = key_part.to_sym
+              iterator[key] = keys[index + 1] ? (iterator[key] || {}) : t.value
+              iterator[key]
+            end
+          end
+          @initialized = true
+        end
+
+        def translations(do_init: false)
```

Comment:
Please remove `do_init`, as we can use `.reload` for that purpose
```ruby
def translations
  @translations ||= Translation.to_hash
end
```
and with adding `return self` in `.reload!`, `do_init` can be expressed as
```ruby
I18n.backend.translations

I18n.backend.reload!.translations
```

Reply:
I agree with you, the whole `do_init` part is needlessly convoluted, but I was trying to implement the same "API" as i18n's `I18n::Backend::Simple`. API is in quotation marks because a few of these are not even marked as public methods, and that encapsulation is broken at several levels by both i18n-js and i18n itself, which use `send` or `instance_eval` to get around the visibility constraint for some reason.

In particular, `translations` gets called as `translations(do_init: true)` by i18n-js. And the `I18n::Backend::Chain` backend runs `init_translations` and `initialized?` on all of its child backends.

So I've essentially empirically determined these methods with these signatures to be part of the actual interface expected of an i18n backend implementation and implemented them here, as unfortunate as they are.

I've changed the implementation of `translations` to initialize by default, so `I18n.backend.reload!.translations` will work, though.

Reply:
Indeed, I can see that [here](https://github.com/ruby-i18n/i18n/blob/60f0830d5a8a78c2218be9e367044e633950fc4a/lib/i18n/backend/key_value.rb#L126-L132) and [here](https://github.com/ruby-i18n/i18n/blob/e7507e4562ce3116695c69ed68393bc4adde0277/lib/i18n/backend/simple.rb#L58-L82)

Well, so let's merge it as it is